### PR TITLE
Fixed `stretched_grid()` function

### PR DESCRIPTION
**Pull request by taimoorsohail**
I've modified the stretched_grid() function so it matches the specified Oceananigans format

### CODE REVIEW

**Comment by navidcy** [comment]
What is `decay`? Why it depends on `Nz`?

**Comment by navidcy** [comment]
Let's use more verbose variable names. For example, `decay_lengthscale`? Although, dimensionally I think `decay` has dimensions of `1 / Length`.

File: asc.jl [hunk 1]
```diff
@@ -19,7 +19,10 @@ Nx, Ny, Nz = 128, 128, 64
 
 decay = Nz/Lz*2
 
-stretched_grid(k) = -(Lz*(tanh(decay*(k-Nz-0.5))))+(Lz*tanh(decay*(-Nz+1-0.5)));
+
+stretched_grid(k) = -(-(Lz*(tanh(decay*(k+Nz+0.5)))-(Lz*tanh(decay*(2*Nz+1.5))))/(-(Lz*(tanh(decay*(Nz+1.5)))-(Lz*tanh(decay*(2*Nz+1.5)))))*Lz)
```
Comment (navidcy) [inline asc.jl:23]:
```suggestion
stretched_grid(k) = Lz * (tanh(decay * (2Nz + 3/2)) - tanh(decay * (Nz + 1/2 + k))) / 
                            (tanh(decay * (Nz + 3/2)) - tanh(decay * (2Nz + 3/2)))
```

tried to clean it up a bit

Comment (navidcy) [reply asc.jl:23]: Also, I suggest after we create the grid we add

```Julia
fig = Figure()
ax = Axis(fig[1, 1], ylabel = "Depth (m)", xlabel = "Vertical spacing (m)")
lines!(ax, grid.Δzᵃᵃᶜ[1:grid.Nz], grid.zᵃᵃᶜ[1:grid.Nz])
scatter!(ax, grid.Δzᵃᵃᶜ[1:Nz], grid.zᵃᵃᶜ[1:Nz])
current_figure()
```

even if we have it commented out! This way one can visualize the stretched grid spacing easily.

**Comment by navidcy** [comment]
Another suggestions is that we rename the `stretched_grid()` to `z_faces()` which is more accurate.

**Comment by txs156** [comment]
> What is `decay`? Why it depends on `Nz`?

`decay` is a coefficient we multiply by the tanh profile that stretches the non-constant part (i.e. the bit that's symmetric about 0). I've found in trial and error that `decay` needs to change based on the number of grid points `Nz`. I'm sure this can be generalised...

File: asc.jl (hunk 1 above)
Comment (txs156) [reply asc.jl]: Done :)

**Pull request merged**